### PR TITLE
fix(hybrid-cloud): Adds defaults to provisioning model fields, lost_password_hash model

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Fixes a few issues that crop up as errors in the Pydantic upgrade work:
- Optional fields without defaults now cause failures instead of assuming a value of None
- Type mismatches between fields are no longer automatically coerced without an additional config setting

### CODE REVIEW

File: src/sentry/users/services/lost_password_hash/model.py [hunk 1]
```diff
@@ -13,7 +16,7 @@ class RpcLostPasswordHash(RpcModel):
     id: int = -1
     user_id: int = -1
     hash: str = ""
-    date_added = datetime.datetime
+    date_added: datetime = Field(default_factory=timezone.now)
```
Comment (GabeVillalobos) [inline src/sentry/users/services/lost_password_hash/model.py:19]: I'm not super sure we want this to be the default. I could also remove this entirely and just assume the caller needs to provide one, but I this was failing tests for whatever reason 🤔

Comment (corps) [reply src/sentry/users/services/lost_password_hash/model.py:19]: I think it's a reasonable default.  Forcing the caller to provide one is reasonable in tests, but not in production, due to the reality that there will be (increasing) version drift between callers and receivers.

Comment (corps) [reply src/sentry/users/services/lost_password_hash/model.py:19]: Whatever default does it exist, it should be safe for the purpose of backwards compatibility, which depends on the use case.

**Review by RyanSkonnord — APPROVED** [review]

File: src/sentry/users/services/lost_password_hash/model.py (hunk 1 above)
Comment (markstory) [reply src/sentry/users/services/lost_password_hash/model.py:19]: This default seems fine to me. Looking at the existing usage, we only create `RpcLostPasswordHash` by serializing an ORM object, which would have `date_added` set as well.

**Review by markstory — APPROVED** [review]

**Pull request merged**